### ISSUE TITLE
Strange behavior of Tapo P110 energy monitoring in Home Assistant

### ISSUE THREAD

**Issue by ElyFNV** [petretiandrea/home-assistant-tapo-p100#286]
Hi @petretiandrea
Forgive me for writing, but I have noticed that since the last firmware update, my tapo p110 smart plugs are having problems reporting power consumption correctly.
Actually I don't know if it's a Tapo or Home Assistant problem. My problem is the following:
I created a sensor monitoring my monthly energy consumption of a specific item using "today_energy" sensor and the "HA utility meter". Everything works perfectly... Or at least I thought.
I realized, in fact, that the value of the sensor created (of the helper created by utility_meter to be clear), turns out to be different if taken individually or if observed from the Home Assistant energy dashboard. And I don't understand how this is possible since the sensor is the same! It is as if HA had 2 different values of the same sensor! It does not make sense! You can see what I mean from the attached pictures.

![1](https://user-images.githubusercontent.com/37159948/204305185-92c1f7cf-28e6-41cb-a86b-42924437eca8.png)
![2](https://user-images.githubusercontent.com/37159948/204305192-f28ef8bc-bf12-43d5-8f73-5ca5953dcdd2.png)

In a nutshell, the value given by the utility_meter changes when it is reported in the energy board. But this argument does not apply to all products connected to Tapo P110. Only with some of them. And I can't explain why!

**Comment by joshuevic** [comment]
@ElyFNV I think the difference is in the selection you made on the energy dashboard. You have selected to give you the total consumption for the current month (Novembre 2022). Instead, the single "Consumo-Lavastoviglie" sensor gives you the total consumption for the day because it's the "today_energy" sensor of the P110. Theoretically that sensor resets itself every day.
